data-analytics/dataflow-python-examples/dataflow_python_examples/data_generation_for_benchmarking.py

```diff
@@ -0,0 +1,485 @@
+                                              self.data_gen.float_precision,
+                                              self.data_gen.only_pos)
+            record[fieldname] = float(record[fieldname])
+
+        # Make some values null based on null_prob.
+        if field[u'mode'] == 'NULLABLE':
+            record[fieldname] = np.random.choice([None, record[fieldname]],
+                                                 p=[self.data_gen.null_prob,
+                                                    1.0 - self.data_gen.null_prob])
+
+        # Pick key at random from foreign keys.
+        # Draw key column from [0, n_keys) if has _key in the name.
+        # This forces key column to no contain nulls
+        if ('_key' in field[u'name'].lower()) or ('_id' in field[u'name'].lower()):
+            key_mag = int(math.log10(self.data_gen.n_keys))
+            # Assume the key field is of string type and format it to be left zero padded.
+            record[fieldname] = format(np.random.randint(0, self.data_gen.n_keys),
+                                       '0' + str(key_mag + 1) + 'd')
+        return record
+
+    def generate_fake(self, fschema):
+        """
+        This method creates a single fake record based on the constraints defined int
+        the FakeRowGen instance's data_gen attribute.
+
+        Arguments:
+                fschema: A dictionary containing a faker_schema (this should be generated by
+                         DataGenerator.get_faker_schema() )
+        """
+        # Initialize a FakerSchema object.
+        schema_faker = FakerSchema()
+
+        # Generate a fake record.
+        data = schema_faker.generate_fake(fschema, 1)  # Generate one record.
+
+        # This performs a sanity check on datatypes and parameterized constraints.
+        for col_name in data:
+            data = self.sanity_check(data, col_name)
+
+        return data
+
+    def process(self, element, *args, **kwargs):
+        """This function creates n random data records based on the properties
+        of the passed DataGenerator object for each element in prior the PCollection.
+
+        Args:
+            element: A single element of the PCollection (the contents of this element are ignored
+                     in this particularDoFn so we just pass it newlines).
+        """
+
+        faker_schema = self.data_gen.get_faker_schema()
+
+        row = [self.generate_fake(faker_schema)]
+        return row  # PCollection returned by DoFn.process() must be an iterable.
+
+
+def run(argv=None):
+    """
+    This funciton parses the command line arguments and runs the Beam Pipeline.
+
+    Args:
+        argv: list containing the commandline arguments for this call of the script.
+    """
+    schema_inferred = False
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--schema_file', dest='schema_file', required=False,
+                        help='Schema json file to read. This can be a local file or a file in a \
```

Comment:
Use built in string concatenation instead of \ for line continuation. Here and on multiple lines below.

```
parser.add_argument('--schema_file', dest='schema_file', required=False,
    help='Schema json file to read. This can be a local file or a file in a '
         'Google Storage Bucket.')
```

Reply:
Correcting line continuation style